app/controllers/api/databases.php

```diff
@@ -115,19 +126,61 @@ function createAttribute(string $databaseId, string $collectionId, Document $att
             'format' => $format,
             'formatOptions' => $formatOptions,
             'filters' => $filters,
+            'options' => $options,
         ]);
 
         $dbForProject->checkAttribute($collection, $attribute);
         $attribute = $dbForProject->createDocument('attributes', $attribute);
-    } catch (DuplicateException $exception) {
+    } catch (DuplicateException) {
         throw new Exception(Exception::ATTRIBUTE_ALREADY_EXISTS);
-    } catch (LimitException $exception) {
+    } catch (LimitException) {
         throw new Exception(Exception::ATTRIBUTE_LIMIT_EXCEEDED, 'Attribute limit exceeded');
     }
 
     $dbForProject->deleteCachedDocument('database_' . $db->getInternalId(), $collectionId);
     $dbForProject->deleteCachedCollection('database_' . $db->getInternalId() . '_collection_' . $collection->getInternalId());
 
+    if ($type === Database::VAR_RELATIONSHIP && $options['twoWay']) {
+        $twoWayKey = $options['twoWayKey'];
+        $options['relatedCollection'] = $collection->getId();
+        $options['twoWayKey'] = $key;
+        $options['side'] = Database::RELATION_SIDE_CHILD;
+
+        try {
+            $twoWayAttribute = new Document([
+                '$id' => ID::custom($db->getInternalId() . '_' . $relatedCollection->getInternalId() . '_' . $twoWayKey),
+                'key' => $twoWayKey,
+                'databaseInternalId' => $db->getInternalId(),
+                'databaseId' => $db->getId(),
+                'collectionInternalId' => $relatedCollection->getInternalId(),
+                'collectionId' => $relatedCollection->getId(),
+                'type' => $type,
+                'status' => 'processing', // processing, available, failed, deleting, stuck
+                'size' => $size,
+                'required' => $required,
+                'signed' => $signed,
+                'default' => $default,
+                'array' => $array,
+                'format' => $format,
+                'formatOptions' => $formatOptions,
+                'filters' => $filters,
+                'options' => $options,
+            ]);
+
+            $dbForProject->checkAttribute($relatedCollection, $twoWayAttribute);
+            $dbForProject->createDocument('attributes', $twoWayAttribute);
+        } catch (DuplicateException) {
+            $dbForProject->deleteDocument('attributes', $attribute->getId());
+            throw new Exception(Exception::ATTRIBUTE_ALREADY_EXISTS);
+        } catch (LimitException) {
+            $dbForProject->deleteDocument('attributes', $attribute->getId());
+            throw new Exception(Exception::ATTRIBUTE_LIMIT_EXCEEDED, 'Attribute limit exceeded');
+        }
+
+        $dbForProject->deleteCachedDocument('database_' . $db->getInternalId(), $relatedCollection->getId());
+        $dbForProject->deleteCachedCollection('database_' . $db->getInternalId() . '_collection_' . $relatedCollection->getInternalId());
```

Comment:
```suggestion
         finally {
              $dbForProject->deleteCachedDocument('database_' . $db->getInternalId(), $relatedCollection->getId());
              $dbForProject->deleteCachedCollection('database_' . $db->getInternalId() . '_collection_' . $relatedCollection->getInternalId());
         }
```

I believe this needs to be added inside a finally block as we would need to delete the cached document and collection if an error occurs as well. 

When an error occurs, the cache could be left in an inconsistent state and the attribute may still exist in the cache which will cause future requests to createAttribute with the same `key` to fail.